ledger/participant-integration-api/src/main/scala/platform/store/backend/common/InitHookDataSourceProxy.scala

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.backend.common
+
+import java.io.PrintWriter
+import java.sql.{
+  Blob,
+  CallableStatement,
+  Clob,
+  Connection,
+  DatabaseMetaData,
+  NClob,
+  PreparedStatement,
+  SQLWarning,
+  SQLXML,
+  Savepoint,
+  Statement,
+  Struct,
+}
+import java.util.Properties
+import java.util.concurrent.Executor
+import java.util.logging.Logger
+import java.{sql, util}
+
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import javax.sql.DataSource
+
+private[backend] object InitHookDataSourceProxy {
+  val logger: ContextualizedLogger = ContextualizedLogger.get(this.getClass)
+
+  def apply(
+      delegate: DataSource,
+      initHooks: List[Connection => Unit],
+  )(implicit loggingContext: LoggingContext): DataSource =
+    if (initHooks.isEmpty) delegate
+    else InitHookDataSourceProxy(delegate, c => initHooks.foreach(_(c)))
+}
+
+import com.daml.platform.store.backend.common.InitHookDataSourceProxy._
+
+private[backend] case class InitHookDataSourceProxy(
+    delegate: DataSource,
+    initHook: Connection => Unit,
+)(implicit loggingContext: LoggingContext)
+    extends DataSource {
+  override def getConnection: Connection = {
+    val connectionId =
+      List.fill(8)(scala.util.Random.nextPrintableChar()).mkString // TODO FIXME maybe not needed
+    logger.debug(s"$connectionId Creating new connection")
+    val connection = delegate.getConnection
+    try {
+      logger.debug(s"$connectionId Applying connection init hook")
+      initHook(connection)
+    } catch {
+      case t: Throwable =>
+        logger.info(s"$connectionId Init hook execution failed: ${t.getMessage}")
```

Comment:
Shouldn't this be a warning?

Reply:
That is a good question: it depends on our logging policies. 
If potentially transient problems should materialize as warnings, then yes.
@mziolekda can you maybe comment on this

Reply:
We are lacking a comprehensive strategy in that respect, but we have plenty of prior art. Transient problems pertaining to platform capabilities have usually been logged as warnings.

Reply:
warning it is, changing it